modules/manifold/src/utils/data-slicing.js

```diff
@@ -0,0 +1,28 @@
+import {dotRange, computeClusters} from '@mlvis/mlvis-common/utils';
+import {gatherDataset, filterDataset} from './utils';
+
+export const computeManualSegmentationResult = (data, segmentFilters) => {
+  return segmentFilters.map(filter => filterDataset(data, filter));
+};
+
+export const computeAutoSegmentationResult = (
+  data,
+  columnTypeRanges,
+  nClusters
+) => {
+  const colIds = dotRange(...columnTypeRanges.score);
+  const clusteringInputDataset = gatherDataset(data, colIds);
+
+  const {columns} = clusteringInputDataset;
+  const clusterIds = computeClusters(columns, nClusters, true);
```

Comment:
not in this diff but what do you think of using one object as the input parameter, instead of three parameters? such that the order of the params won't matter and it is easier to infer the keys (e.g. what are we putting the `true` for here?)

Reply:
Sure, `true` was an indicator of whether input shape is [nDataPoints, nFeatures] or [nFeatures, nDataPoints]. Was added as a quick fix when we did the refactoring to migrate to column based data format. Will update `computeClusters` so we don't have to use `true`